From 8cb9cdede367e1c4c4efb968d7a61b2274bb666e Mon Sep 17 00:00:00 2001
From: Thomas Backlund <tmb@mageia.org>
Date: Sun, 8 Apr 2012 23:17:20 +0100
Subject: [PATCH 303/303] Read *.xsetup scripts.

This patch restores the sourcing/executing of *.xsetup scripts
in gdm from gdm-2.29.5-defaultconf.patch used in Mageia 1.

Without this atleast Gnome LiveCDs are broken.
---
 data/Init.in | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/data/Init.in b/data/Init.in
index ec6fd39..0b88213 100644
--- a/data/Init.in
+++ b/data/Init.in
@@ -86,4 +86,17 @@ if [ "x$SETXKBMAP" != "x" ] ; then
   fi
 fi
 
+# NOTE! LiveCDs and other rpms expects xsetup scripts to be executed by gdm
+for i in /etc/X11/xsetup.d/*.xsetup ; do
+    [ -d $i ] && continue
+
+    if [ -x $i ]; then
+	if grep -q "# to be sourced" $i; then
+	    . $i
+	else
+	    $i &
+	fi
+    fi
+done
+
 exit 0
-- 
1.7.11.4

